prov/util/src/util_mem_hooks.c

```diff
@@ -112,7 +113,12 @@ struct ofi_intercept {
 	const char			*symbol;
 	void				*our_func;
 	void				*orig_func;
+#if (defined(__riscv) && (__riscv_xlen == 64))
+	unsigned char			patch_data[OFI_INTERCEPT_MAX_PATCH]
+		                        __attribute__ ((aligned(32)));
+#else
 	unsigned char			patch_data[OFI_INTERCEPT_MAX_PATCH];
+#endif
```

Comment:
Could we set attribute(aligned)) always and drop the ifdef?  Is there any real downside to doing that?

Reply:
@shefty the attribute is there for riscv ABI reasons; the software works without the attribute so it can be removed if that's the preference.

Reply:
I was suggesting the opposite.  Just add it without the ifdef.

Reply:
@shefty all set, done!